hip/test/matrix/coo_kernels.hip.cpp

```diff
@@ -0,0 +1,242 @@
+        mtx = Mtx::create(ref);
+        mtx->copy_from(gen_mtx(532, 231));
+        expected = gen_mtx(532, num_vectors);
+        y = gen_mtx(231, num_vectors);
+        alpha = gko::initialize<Vec>({2.0}, ref);
+        beta = gko::initialize<Vec>({-1.0}, ref);
+        dmtx = Mtx::create(hip);
+        dmtx->copy_from(mtx.get());
+        dresult = Vec::create(hip);
+        dresult->copy_from(expected.get());
+        dy = Vec::create(hip);
+        dy->copy_from(y.get());
+        dalpha = Vec::create(hip);
+        dalpha->copy_from(alpha.get());
+        dbeta = Vec::create(hip);
+        dbeta->copy_from(beta.get());
+    }
+
+
+    std::shared_ptr<gko::ReferenceExecutor> ref;
+    std::shared_ptr<const gko::HipExecutor> hip;
+
+    std::ranlux48 rand_engine;
+
+    std::unique_ptr<Mtx> mtx;
+    std::unique_ptr<Vec> expected;
+    std::unique_ptr<Vec> y;
+    std::unique_ptr<Vec> alpha;
+    std::unique_ptr<Vec> beta;
+
+    std::unique_ptr<Mtx> dmtx;
+    std::unique_ptr<Vec> dresult;
+    std::unique_ptr<Vec> dy;
+    std::unique_ptr<Vec> dalpha;
+    std::unique_ptr<Vec> dbeta;
+};
+
+
+TEST_F(Coo, SimpleApplyIsEquivalentToRef)
+{
+    set_up_apply_data();
+
+    mtx->apply(y.get(), expected.get());
+    dmtx->apply(dy.get(), dresult.get());
+
+    GKO_ASSERT_MTX_NEAR(dresult, expected, 1e-14);
+}
+
+
+// Comment it until finishing the hip dense kernel
+// TEST_F(Coo, AdvancedApplyIsEquivalentToRef)
+// {
+//     set_up_apply_data();
+
+//     mtx->apply(alpha.get(), y.get(), beta.get(), expected.get());
+//     dmtx->apply(dalpha.get(), dy.get(), dbeta.get(), dresult.get());
+
+//     GKO_ASSERT_MTX_NEAR(dresult, expected, 1e-14);
+// }
+
+
+TEST_F(Coo, SimpleApplyAddIsEquivalentToRef)
+{
+    set_up_apply_data();
+
+    mtx->apply2(y.get(), expected.get());
+    dmtx->apply2(dy.get(), dresult.get());
+
+    GKO_ASSERT_MTX_NEAR(dresult, expected, 1e-14);
+}
+
+
+TEST_F(Coo, AdvancedApplyAddIsEquivalentToRef)
+{
+    set_up_apply_data();
+
+    mtx->apply2(alpha.get(), y.get(), expected.get());
+    dmtx->apply2(dalpha.get(), dy.get(), dresult.get());
+
+    GKO_ASSERT_MTX_NEAR(dresult, expected, 1e-14);
+}
+
+
+TEST_F(Coo, SimpleApplyToDenseMatrixIsEquivalentToRef)
+{
+    set_up_apply_data(3);
+
+    mtx->apply(y.get(), expected.get());
+    dmtx->apply(dy.get(), dresult.get());
+
+    GKO_ASSERT_MTX_NEAR(dresult, expected, 1e-14);
+}
+
+
+// Comment it until finishing the hip dense kernel
+// TEST_F(Coo, AdvancedApplyToDenseMatrixIsEquivalentToRef)
+// {
+//     set_up_apply_data(3);
+
+//     mtx->apply(alpha.get(), y.get(), beta.get(), expected.get());
+//     dmtx->apply(dalpha.get(), dy.get(), dbeta.get(), dresult.get());
+
+//     GKO_ASSERT_MTX_NEAR(dresult, expected, 1e-14);
+// }
+
+
+TEST_F(Coo, SimpleApplyAddToDenseMatrixIsEquivalentToRef)
+{
+    set_up_apply_data(3);
+
+    mtx->apply2(y.get(), expected.get());
+    dmtx->apply2(dy.get(), dresult.get());
+
+    GKO_ASSERT_MTX_NEAR(dresult, expected, 1e-14);
+}
+
+
```

Comment:
What about `SimpleApplyAddToLargeDenseMatrixIsEquivalentToRef`? Maybe add it again? (commented out)

Reply:
Nevermind, that is taken care of in pull request #373

Reply:
Good catch. It should be here, and it has been in the last commit